cider/exec/nextgen/jitlib/llvmjit/LLVMJITValue.cpp

```diff
@@ -186,7 +186,17 @@ JITValuePointer LLVMJITValue::sub(JITValue& rh) {
 
 JITValuePointer LLVMJITValue::add(JITValue& rh) {
   LLVMJITValue& llvm_rh = static_cast<LLVMJITValue&>(rh);
-  checkOprandsType(this->getValueTypeTag(), rh.getValueTypeTag(), "add");
+
+  if (JITTypeTag::POINTER == getValueTypeTag()) {
+    auto rh_type = rh.getValueTypeTag();
```

Comment:
if we defined JITTypeTag as
```c++
enum class JITTypeTag : int {
  ...
  INT8 = 0x80,
  INT16 = 0x100,
  INT32 = 0x200,
  INT64 = 0x400,
  ...
};
```
and define a const value `constexpr int int_type = JITTypeTag::INT8 || JITTypeTag::INT16 || JITTypeTag::INT32 || JITTypeTag::INT64;`
then following code could be write: 
```c++
if (rh_type & int_type == 0) {
  LOG(FATAL) << ...;
} else {
  checkOprandsType(...);
}
```